samples/build-duration-logger/index.html

```diff
@@ -10,28 +10,63 @@
     const gradleEnterpriseServer = 'https://gradle.my-company.com'; // replace with your Gradle Enterprise base url
     const buildStreamUrl = `${gradleEnterpriseServer}/build-export/v1/builds/since/now?stream`;
     const buildStream = new EventSource(buildStreamUrl);
-    const buildEventStreamUrl = buildId => `${gradleEnterpriseServer}/build-export/v1/build/${buildId}/events?eventTypes=BuildStarted,BuildFinished`;
+
+    class BuildProcessor {
+
+        constructor(gradleEnterpriseServer, maxConcurrentBuildsToProcess) {
+            this.gradleEnterpriseServer = gradleEnterpriseServer;
+            this.buildIds = [];
+            this.numBuildsInProcess = 0;
+            this.maxConcurrentBuildsToProcess = maxConcurrentBuildsToProcess;
+        }
+
+        enqueue(buildId) {
+            this.buildIds.push(buildId);
+            this.processQueuedBuilds();
+        }
+
+        processQueuedBuilds() {
```

Comment:
I would call this `processPendingBuilds()` to be consistent with `pendingBuildIds`.

Reply:
Done